tests/func/test_repro.py

```diff
@@ -1372,210 +1373,199 @@ def test(self):
     """
-    os.unlink(repro_dir.FOO)
-    shutil.copyfile(repro_dir.BAR, repro_dir.FOO)
+    origin_data = Path("origin_data")
+    origin_data.unlink()
+    origin_data.write_text("new origin content")
 
-    os.unlink(repro_dir.DATA)
-    shutil.copyfile(repro_dir.BAR, repro_dir.DATA)
+    dir_file = repro_dir / "data_dir" / "dir_file"
+    dir_file.unlink()
+    dir_file.write_text("new dir file content")
 
-    stages = dvc_repo.reproduce(
-        repro_dir.dname, recursive=True, single_item=True
-    )
+    stages = dvc.reproduce("dir", recursive=True, single_item=True)
     # Check that just stages inside given dir
     # with changed direct deps have been reproduced.
     # This means that "first" stage should not be reproduced
     # since it depends on "source".
-    # Also check that "second" stage was reproduced before "third" stage
+    # Also check that "dir_file_copy" stage was reproduced before "third" stage
     assert len(stages) == 2
-    assert repro_dir.second_stage == stages[0].relpath
-    assert repro_dir.third_stage == stages[1].relpath
-    assert filecmp.cmp(repro_dir.second, repro_dir.BAR, shallow=False)
+    assert [s.relpath for s in stages] == [
+        repro_dir.dir_file_copy + ".dvc",
+        repro_dir.last_stage,
+    ]
+    assert Path(repro_dir.dir_file_copy).read_text() == "new dir file content"
 
 
-def test_recursive_repro_single_force(dvc_repo, repro_dir):
+def test_recursive_repro_single_force(dvc, repro_dir):
     """
     Test recursive single-item force repro on dir
     without any dependencies changing.
     """
-    stages = dvc_repo.reproduce(
-        repro_dir.dname, recursive=True, single_item=True, force=True
-    )
+    stages = dvc.reproduce("dir", recursive=True, single_item=True, force=True)
     assert len(stages) == 3
-    names = [stage.relpath for stage in stages]
     # Check that all stages inside given dir have been reproduced
-    # Also check that "second" stage was reproduced before "third" stage
+    # Also check that "dir_file_copy" stage was reproduced before "third" stage
     # and that "first" stage was reproduced before "third" stage
-    assert repro_dir.first_stage in names
-    assert repro_dir.second_stage in names
-    assert repro_dir.third_stage in names
-    assert names.index(repro_dir.first_stage) < names.index(
-        repro_dir.third_stage
+    names = [s.relpath for s in stages]
+    assert {
+        repro_dir.origin_copy_2 + ".dvc",
+        repro_dir.dir_file_copy + ".dvc",
+        repro_dir.last_stage,
+    } == set(names)
+    assert names.index(repro_dir.origin_copy_2 + ".dvc") < names.index(
+        repro_dir.last_stage
     )
-    assert names.index(repro_dir.second_stage) < names.index(
-        repro_dir.third_stage
+    assert names.index(repro_dir.dir_file_copy + ".dvc") < names.index(
+        repro_dir.last_stage
     )
 
 
-def test_recursive_repro_empty_dir(dvc_repo, repro_dir):
+def test_recursive_repro_empty_dir(dvc, repro_dir):
```

Comment:
I presume that `repro_dir` was added here to make sure running `repro`, will not trigger any other stage?

Reply:
@pared I think it was only used for `repro_dir.emtydname`, so we can safely remove it.

Reply:
@efiop removed